source/common/network/socket_option_impl.cc

```diff
@@ -34,6 +36,26 @@ bool SocketOptionImpl::setOption(Socket& socket, Socket::SocketState state) cons
     }
   }
 
+  if (state == Socket::SocketState::Listening) {
+    if (tcp_fast_open_queue_length_.has_value()) {
+      const int tfo_value = tcp_fast_open_queue_length_.value();
+      const SocketOptionName option_name = ENVOY_SOCKET_TCP_FASTOPEN;
+      if (option_name) {
+        const int error = Api::OsSysCallsSingleton::get().setsockopt(
```

Comment:
You should provide a TCP equivalent of https://github.com/envoyproxy/envoy/blob/master/source/common/network/socket_option_impl.h#L69.

Reply:
The IP options are more complicated because we have to deal with different options, only sometimes, between ipv4 and ipv6. AFAIK that isn't a thing with tcp socket options, and there's currently only 1 of them. Still think we should add a function for it at this time?

Reply:
I'd suggest to still do this, since we factor out the "Unsupported socket option .." detection and logging code. I think there's a bug in the code here as is; if the socket option is not supported, in this PR it just logs and returns true. It should  log and return false (due to the ENOTSUP, which we would generate in the suggested `setTcpSocketOption`).

Reply:
I still have to do this..